src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -858,10 +896,21 @@ private void FleeFromPredators(ref WorldPosition position, ref MicrobeAI ai, ref
     }
 
     private void EngagePrey(ref MicrobeAI ai, ref MicrobeControl control, ref OrganelleContainer organelles,
-        ref WorldPosition position, CompoundBag ourCompounds, in Entity entity, Vector3 target, bool engulf,
-        float speciesAggression, float speciesFocus, float speciesActivity, float strain, Random random)
+        ref WorldPosition position, ref CompoundStorage compoundStorage, ref Health health, in Entity entity,
+        Vector3 target, bool engulf, float speciesAggression, float speciesFocus, float speciesActivity,
+        float strain, Random random, float atpLevel)
     {
-        control.SetStateColonyAware(entity, engulf ? MicrobeState.Engulf : MicrobeState.Normal);
+        var ourCompounds = compoundStorage.Compounds;
+
+        if (atpLevel <= Constants.ENGULF_NO_ATP_TRIGGER_THRESHOLD)
```

Comment:
I see the `EnterEngulfModeForcedState` call now, but now I'm wondering why the ATP level is checked here?  Because `EnterEngulfModeForcedState` ultimately calls `ForceStateApplyIfRequired` which only causes problems if ATP is not sufficient, so the AI code doesn't actually need this duplicate check, right?

Reply:
So we just use `EnterEngulfModeForcedState` withour `SetStateColonyAware`?

Reply:
Yes, I didn't fully read the PR again but from a quick look `EnterEngulfModeForcedState` already has colony handling and checking for the ATP levels. So my comment is just about reducing unnecessary code in the AI system in this spot.